### PR TITLE
Stealing keys correctly handles map expressions

### LINKED ISSUE #3: Issue with UltiSnips snippets
I have a snippet that expands
```c
if
```
to
```c
if (true) {
	
}
```
However, when I also use deoplete-clang it expands to
```c
if<Plug>delimitMateS-Tab
```
instead. The problem persists regardless of my UltiSnips settings, and it doesn't happen with deoplete-clang.

### PR DESCRIPTION
**Pull request by tweekmonster**
Fixes #3 

@HiPhish could you see if this resolves your issue?

### CODE REVIEW

**Comment by HiPhish** [comment]
Now pressing `<TAB>` doesn't do anything anymore.

**Comment by tweekmonster** [comment]
Perfect!  Working as intended.  😀

I guess I'll have to take a closer look at what delimitMate is actually doing.

**Comment by tweekmonster** [comment]
@HiPhish Try it again.  Please also confirm that `<tab>` and `<s-tab>` cycles through `<#templates#>` correctly, and `]` actually inserts `]`.

File: autoload/clang2.vim [hunk 1]
```diff
@@ -150,28 +150,28 @@ endfunction
...
+        return saved_key
+      endif
+    endif
+
+    if a:dir != 0
+      let orig_key = a:dir == -1 ? s:pl_prev : s:pl_next
+      if orig_key =~# '^<[^<>]*>$'
+        return eval('"\'.orig_key.'"')
+      endif
+      return orig_key
```
Comment (tweekmonster) [inline autoload/clang2.vim]: Bad logic above.  Will fix soon.

**Comment by HiPhish** [comment]
Now pressing `<TAB>` only inserts a tab character, it doesn't expand the snippet. But the function argument templates do work.

**Comment by tweekmonster** [comment]
Alright, it was a missing `!` that fooled me into thinking I fixed things.  It uncovered a bunch of other problems.  It should work now.

**Comment by HiPhish** [comment]
It works now 👍

**Pull request merged**